nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/test/java/org/apache/nifi/jms/cf/JMSConnectionFactoryProviderTest.java

```diff
@@ -101,4 +101,157 @@ public void validateGetConnectionFactoryFailureIfServiceNotConfigured() throws E
         new JMSConnectionFactoryProvider().getConnectionFactory();
     }
 
+    @Test
+    public void validWithSingleTestBroker() throws Exception {
+        TestRunner runner = TestRunners.newTestRunner(mock(Processor.class));
+
+        JMSConnectionFactoryProvider cfProvider = new JMSConnectionFactoryProvider();
+        runner.addControllerService("cfProvider", cfProvider);
+
+        String clientLib = this.getClass().getResource("/dummy-lib.jar").toURI().toString();
+
+        runner.setProperty(cfProvider, JMSConnectionFactoryProvider.BROKER_URI, "myhost:1234");
+        runner.setProperty(cfProvider, JMSConnectionFactoryProvider.CLIENT_LIB_DIR_PATH, clientLib);
+        runner.setProperty(cfProvider, JMSConnectionFactoryProvider.CONNECTION_FACTORY_IMPL,
+                "org.apache.nifi.jms.testcflib.TestConnectionFactory");
+
+        runner.assertValid(cfProvider);
```

Comment:
Since JMSConnectionFactoryProvider doesn't implement `customValidate` method and the NonEmptyBrokerURIValidator used for validating the BROKER_URI property doesn't check connection factory impl value, these test variations do not assert what is expected.

Alternative approach would be creating a sub-class of JMSConnectionFactoryProvider only for testing purpose which overrides `setProperty(String propertyName, Object propertyValue)` then do assertion within the overridden setProperty method. This way, we can make sure what property is set by combination of broker uri and connection factory impl.

How do you think?

Reply:
Thanks for your valuable comment. My initial intent was to unit test the `setProperty(String propertyName, Object propertyValue)` method but realized that the tests did not execute this method. I kept the tests despite this realization with the purpose to ensure that future changes to validation will accept these broker URIs. 

NIce idea with the sub-class! I will try to implement this